### PR TITLE
Fix Grails application error

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: instrumentation/grails-3.0/javaagent/src/test/groovy/test/GrailsTest.groovy [hunk 1]
```diff
@@ -56,8 +57,10 @@ class GrailsTest extends HttpServerTest<ConfigurableApplicationContext> implemen
       return getContextPath() + "/test/path"
     } else if (endpoint == QUERY_PARAM) {
       return getContextPath() + "/test/query"
-    } else if (endpoint == ERROR || endpoint == EXCEPTION) {
+    } else if (endpoint == ERROR) {
       return getContextPath() + "/error/index"
+    } else if (endpoint == EXCEPTION) {
+      return getContextPath() + "/error"
```
Comment (trask) [inline instrumentation/grails-3.0/javaagent/src/test/groovy/test/GrailsTest.groovy:63]: this change is being overwritten in #2814

**Comment by iNikem** [comment]
@trask what are your plans for this PR?

**Comment by trask** [comment]
I'm hoping you will accept the large wall of explanatory text in lieu of a test 😭

File: instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerMappingResourceNameFilter.java [hunk 2]
```diff
@@ -74,8 +76,39 @@ private boolean findMapping(HttpServletRequest request) throws Exception {
...
+    for (HandlerMapping mapping : mappings) {
+      // it may be enticing to add all HandlerMapping classes here, but DO NOT
+      //
+      // because we call getHandler() on them above, at the very beginning of the request
+      // and this can be a very invasive call with application-crashing side-effects
+      //
+      // for example: org.grails.web.mapping.mvc.UrlMappingsHandlerMapping.getHandler()
+      // 1. uses GrailsWebRequest.lookup() to get GrailsWebRequest bound to thread local
+      // 2. and populates the servlet request attribute "org.grails.url.match.info"
+      //    with GrailsControllerUrlMappingInfo
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerMappingResourceNameFilter.java:90]: 🤯 

Ah, side effects ❤️

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**